### DISCUSSION TITLE
Choppy Camera while walking

### DISCUSSION THREAD

**Discussion by scope9** [Sixze/ALS-Refactored#565] [Q&A]
Whenever I walk on the ground or stairs the camera is very choppy but the camera for running works fine. Any suggestions to fix this?

https://github.com/user-attachments/assets/51f8fbd2-5839-415a-86d9-75212a1fda5c

**Comment by jomath229** [comment]
I don't really know how the camera system works since I've been using my own, but from what I can see it seems like the linecast for the camera collision on world is starting from the bottom of the character's capsule, since we can see it moving back to its correct position from the capsule's bottom. Since it's starting from down there the linecast is constantly hitting the ground, moving the camera there, since it thinks it's a wall. I also don't know how you've setup your projet, but I'd look into finding the starting point of that linecast and try to move it up so the linecast stops constantly hitting the floor.

**Comment by GStreetGames** [comment]
The camera is colliding with something, either your character or the environment. Look at your collision for these things and alter them to ignore it or have it ignore them.

**Comment by Sixze** [comment]
Make sure that your character's skeleton has all the necessary sockets for the camera and that they are specified in `CS_Als_Default`.
